Assets/Global/Scripts/UIScripts/Healthbar.cs

```diff
@@ -4,19 +4,40 @@
 
 public class Healthbar : MonoBehaviour
 {
-    private Image healthbarOverlay;
-    private Image healthbarUnderlay;
+    private Transform HealthbarUnderlayTransform;
+    private Transform HealthbarOverlayTransform;
+    private float originalWidth;
     // Start is called once before the first execution of Update after the MonoBehaviour is created
-    void Start()
+    void Awake()
     {
-        healthbarOverlay = transform.GetChild(1).GetComponent<Image>();
-        healthbarUnderlay = transform.GetChild(0).GetComponent<Image>();
+        HealthbarUnderlayTransform = transform.GetChild(0);
+        HealthbarOverlayTransform = transform.GetChild(1);
+
+        originalWidth = (HealthbarUnderlayTransform as RectTransform).sizeDelta.x;
+
+        UpdateMaxHealth();
     }
 
-    public void UpdateHealthBar(float min, float max, float current)
+    public void UpdateHealthBar(float current)
```

Comment:
To prevents bugs and stuff, i consistant with your implementation. 
So either:
1. use parameter in the method to get the current and max health.
or
2. use global references to get the current and max health

Same goes for the methods implementation. 
Chose to either 
1. do both the current and max health in 1 method `UpdateHealthBar` 
 Or
 2.split them up in to 2 methods `UpdateMaxHealth` and `UpdateCurrentHealth`